sql/expression/function/date_format.go

```diff
@@ -0,0 +1,316 @@
+	if day < 4 || day > 20 {
+		switch day % 10 {
+		case 1:
+			suffix = "st"
+		case 2:
+			suffix = "nd"
+		case 3:
+			suffix = "rd"
+		}
+	}
+
+	return strconv.FormatInt(day, 10) + suffix
+}
+
+func dayOfMonth(t time.Time) string {
+	return strconv.FormatInt(int64(t.Day()), 10)
+}
+
+func microsecondsStr(t time.Time) string {
+	micros := t.Nanosecond()/int(time.Microsecond)
+	return fmt.Sprintf("%06d", micros)
+}
+
+func minutesStr(t time.Time) string {
+	return fmt.Sprintf("%02d", t.Minute())
+}
+
+func twelveHour(t time.Time) (int, string) {
+	ampm := "AM"
+	if t.Hour() >= 12 {
+		ampm = "PM"
+	}
+
+	hour := t.Hour() % 12
+	if hour == 0 {
+		hour = 12
+	}
+
+	return hour, ampm
+}
+
+func twelveHourPadded(t time.Time) string {
+	hour, _ := twelveHour(t)
+	return fmt.Sprintf("%02d", hour)
+}
+
+func twelveHourNoPadding(t time.Time) string {
+	hour, _ := twelveHour(t)
+	return fmt.Sprintf("%d", hour)
+}
+
+func twentyFourHourNoPadding(t time.Time) string {
+	return fmt.Sprintf("%d", t.Hour())
+}
+
+func fullMonthName(t time.Time) string {
+	 return t.Month().String()
+}
+
+func ampmClockStr(t time.Time) string {
+	hour, ampm :=twelveHour(t)
+	return fmt.Sprintf("%02d:%02d:%02d %s", hour, t.Minute(), t.Second(), ampm)
+}
+
+func secondsStr(t time.Time) string {
+	return fmt.Sprintf("%02d", t.Second())
+}
+
+func yearWeek(mode int32, t time.Time) (int32, int32) {
+	yw := YearWeek{expression.NewLiteral(t, sql.Datetime), expression.NewLiteral(mode, sql.Int32)}
+	res, _ := yw.Eval(nil, nil)
+	yr := res.(int32) / 100
+	wk := res.(int32) % 100
+
+	return yr, wk
+}
+
+func weekMode0(t time.Time) string {
+	yr, wk := yearWeek(0, t)
+
+	if yr < int32(t.Year()) {
+		wk = 0
+	} else if yr > int32(t.Year()) {
+		wk = 53
+	}
+
+	return fmt.Sprintf("%02d", wk)
+}
+
+func weekMode1(t time.Time) string {
+	yr, wk := yearWeek(1, t)
+
+	if yr < int32(t.Year()) {
+		wk = 0
+	} else if yr > int32(t.Year()) {
+		wk = 53
+	}
+
+	return fmt.Sprintf("%02d", wk)
+}
+
+func weekMode2(t time.Time) string {
+	_, wk := yearWeek(2, t)
+	return fmt.Sprintf("%02d", wk)
+}
+
+func weekMode3(t time.Time) string {
+	_, wk := yearWeek(3, t)
+	return fmt.Sprintf("%02d", wk)
+}
+
+func yearMode0(t time.Time) string {
+	yr, _ := yearWeek(0, t)
+	return strconv.FormatInt(int64(yr), 10)
+}
+
+func yearMode1(t time.Time) string {
+	yr, _ := yearWeek(1, t)
+	return strconv.FormatInt(int64(yr), 10)
+}
+
+func dayName(t time.Time) string {
+	return t.Weekday().String()
+}
+
+func yearTwoDigit(t time.Time) string {
+	return strconv.FormatInt(int64(t.Year()) % 100, 10)
+}
+
+type AppendFuncWrapper struct {
+	fn func(time.Time) string
+}
+
+func wrap(fn func(time.Time) string) strftime.Appender {
+	return AppendFuncWrapper{fn}
+}
+
+func (af AppendFuncWrapper) Append(bytes []byte, t time.Time) []byte {
+	s := af.fn(t)
+	return append(bytes, []byte(s)...)
+}
+
+var mysqlDateFormatSpec = strftime.NewSpecificationSet()
+var specifierToFunc = map[byte]func(time.Time) string {
+	'a': nil,
```

Comment:
nil will remove strftime's default behavior for a specifier

Reply:
Are these all of them? Not all letter cases are represented here